cmd/gitops-server/cmd/cmd.go

```diff
@@ -127,13 +130,26 @@ func runCmd(cmd *cobra.Command, args []string) error {
 
 	oidcConfig := options.OIDC
 
+	if options.OIDCSecret != auth.DefaultOIDCAuthSecretName {
+		log.V(logger.LogLevelDebug).Info("Reading OIDC configuration from alternate secret", "secretName", options.OIDCSecret)
+	}
+
 	// If OIDC auth secret is found prefer that over CLI parameters
 	var secret corev1.Secret
 	if err := rawClient.Get(cmd.Context(), client.ObjectKey{
 		Namespace: v1alpha1.DefaultNamespace,
-		Name:      auth.OIDCAuthSecretName,
+		Name:      options.OIDCSecret,
 	}, &secret); err == nil {
+		if len(options.OIDC.ClientSecret) > 0 && len(secret.Data["clientSecret"]) > 0 {
+			log.V(logger.LogLevelWarn).Info("OIDC client configured by both CLI and secret. CLI values will be overridden.")
+		}
 		oidcConfig = auth.NewOIDCConfigFromSecret(secret)
+	} else if err != nil {
+		log.V(logger.LogLevelDebug).Info("Could not read OIDC secret", "secretName", options.OIDCSecret, "error", err)
+	}
+
+	if len(oidcConfig.ClientSecret) > 0 {
```

Comment:
It's usually easier to read `oidcConfig.ClientSecret != ""`

It's easier to understand what ClientSecret is (and they compile to the same code internally), and this is the style used in Effective Go https://go.dev/doc/effective_go